client/admin/components/imagesGridList.jsx

```diff
@@ -0,0 +1,124 @@
+import React from 'react';
+import GridListTile from '@material-ui/core/GridListTile';
+import MuiGridList from '@material-ui/core/GridList';
+import { makeStyles } from '@material-ui/core';
+import { useWidth } from '../hooks';
+
+const CELL_HEIGHT = 200;
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    padding: 4,
+  },
+  gridList: {
+    width: '100%',
+    margin: '0px !important',
+  },
+  placeholder: {
+    backgroundColor: theme.palette.grey[300],
+    height: '100%',
+  },
+  imageWrapper: {
+    borderWidth: '2px',
+  },
+  imageActive: {
+    borderStyle: 'dashed',
+    borderColor: 'black',
+  },
+}));
+
+/**
+ * @param {Object} props
+ * @param {Object[]} props.images
+ * @param {string} props.images.src
+ * @param {string} props.images.id
+ * @param {Set<string>} props.selectedImages
+ * @param {function(Set<string>):void} props.setSelectedImages
+ * @param {number} props.columns
+ * @param {function(string):any} [props.onSelect]
+ * @param {boolean} [props.multiple]
+ */
+export const ImagesGridList = props => {
+  const {
+    images,
+    selectedImages, // Probably Set is not the best solution, especially for single image select
+    setSelectedImages,
+    onSelect = () => {},
+    multiple = true,
+    columns,
+  } = props;
+  const classes = useStyles();
+
+  const newSet = new Set(selectedImages.values());
+
+  const handleImageClick = imageId => {
+    if (newSet.has(imageId)) {
+      newSet.delete(imageId);
```

Comment:
state should never be mutated, if you change Set to an object (`{id: image}` map) it's going to be easier to perform these operations